assets/js/just-the-docs.js

```diff
@@ -479,6 +479,49 @@ jtd.onReady(function(){
   scrollNav();
 });
 
+// Copy button on code
+
+
+{%- if site.copy_button_on_code != false %}
+
+jtd.onReady(function(){
+
+  var codeBlocks = document.querySelectorAll('div.highlighter-rouge, div.listingblock, figure.highlight');
+
+  var svgCopied =  '<svg viewBox="0 0 24 24" class="copy-icon"><use xlink:href="#svg-copied"></use></svg>';
+  var svgCopy =  '<svg viewBox="0 0 24 24" class="copy-icon"><use xlink:href="#svg-copy"></use></svg>';
+
+  codeBlocks.forEach(codeBlock => {
+    var copyButton = document.createElement('button');
+    var timeout = null;
+    copyButton.type = 'button';
+    copyButton.ariaLabel = 'Copy code to clipboard';
+    copyButton.innerHTML = svgCopy;
+    codeBlock.append(copyButton);
+
+    copyButton.addEventListener('click', function () {
+      if(timeout === null) {
+        var code = codeBlock.querySelector('code').innerText.trim();
+        window.navigator.clipboard.writeText(code);
+
+        copyButton.innerHTML = svgCopied;
+
+        var fourSeconds = {{ site.copy_button_on_code_time | default: 4000 }};
```

Comment:
This variable name can be misleading if a user sets it to anything different from 4s? Maybe `copyButtonTimeout` or `timeoutSetting` would be better?